Tests/SonarScanner.MSBuild.Tasks.UnitTest/WriteTelemetryTests.cs

```diff
@@ -0,0 +1,142 @@
+﻿/*
+ * SonarScanner for .NET
+ * Copyright (C) 2016-2025 SonarSource SA
+ * mailto: info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.Build.Utilities;
+using NSubstitute;
+
+namespace SonarScanner.MSBuild.Tasks.UnitTest;
+
+[TestClass]
+public class WriteTelemetryTests
+{
+    public TestContext TestContext { get; set; }
+
+    [TestMethod]
+    public void AllTelemetryIsWrittenToFile()
```

Comment:
Maybe add also a test with other system types for values? 
Is it possible in the context of tasks?

EDIT: ignore me - From what I saw they can only be strings, only Version is supported as non string.